apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -294,6 +294,13 @@ def prepare_tag(
             )
 
         tag, depends = get_graph_info(html_tags)
+        spec_char_list = list("*&^$@!~[]{}()|:@?/\\")
+        invalid_chars = [c for c in spec_char_list if c in tag]
+        if len(invalid_chars) != 0:
```

Comment:
```suggestion
        if SPEC_CHAR_SET & frozenset(tag):
```
A more elegant check is possible if using the above suggestion.

Reply:
I don't really agree this is more elegant, the `&` operator is so infrequently used in Python people don't actually remember what it does.

Reply:
Agreed that it's a little infrequent, and using the fully spelled out `intersection` is more explicit: https://www.programiz.com/python-programming/methods/set/intersection

The main point of my comment wasn't really about the syntax as much as using the built-in set intersection instead of looping.

Reply:
I've changed it to use the frozenset and intersection method.